src/runtime/naming/name.cpp

```diff
@@ -239,6 +241,35 @@ namespace hpx { namespace naming
 
         ///////////////////////////////////////////////////////////////////////
         // prepare the given id, note: this function modifies the passed id
+        void handle_credit_splitting(
+            serialization::output_archive& ar, id_type_impl& gid,
+            serialization::detail::preprocess_gid_types* split_gids)
+        {
+            auto* handle_futures = ar.try_get_extra_data<
+                serialization::detail::preprocess_futures>(
+                serialization::extra_output_handle_futures);
+
+            if (handle_futures == nullptr)
+                return;
+
+            if (split_gids != nullptr)
+            {
+                auto f = split_gid_if_needed(gid).then(hpx::launch::sync,
+                    [split_gids, &gid](hpx::future<gid_type>&& gid_future) {
+                        split_gids->add_gid(gid, gid_future.get());
+                    });
+
+                handle_futures->await_future(
+                    *traits::future_access<decltype(f)>::get_shared_state(f));
+            }
+            else
+            {
+                auto f = split_gid_if_needed(gid);
+                handle_futures->await_future(
+                    *traits::future_access<decltype(f)>::get_shared_state(f));
```

Comment:
This is definitely not correct and will lead to memory leaks if the gid was split.

Reply:
Right. I think we can just throw (assert?) if we get to this point as it shouldn't even happen in the first place.

Reply:
For now, I suggest to not change the current behavior (an assert to catch potential mistakes might do it for now just as well).

Reply:
That's fixed now.